eng/scripts/CodeChecks.ps1

```diff
@@ -5,6 +5,9 @@ param (
     [Parameter(Position=0)]
     [string] $ServiceDirectory,
 
+    [Parameter()]
```

Comment:
I'm not sure I understand why we need to add this parameter. I was hoping we would actually be able to eliminate it everywhere after this work is done.

Reply:
I don't want to put "remove $ServiceToTest" task into this PR.
Here is the reason:
CodeCheck.ps1 includes the following 4 tasks: 
(Note: $ServiceToTask is * for core/releasemanagerment, otherwise $ServiceDirectory)
```
1. Re-generating clients (dotnet msbuild). We use $ServiceDirectory for the PR
2. Re-generating tests (dotnet msbuild). We use $ServiceToTask for the PR
3. Re-generating snippets (script). We use $ServiceToTask for the PR
4. Re-generating listings (script). We use $ServiceToTask for the PR
```
To remove $ServiceToTask, we have to take care of the last two which either run everything in the splitting job or let them run on the Prop file. Both need some efforts to make them run properly.

I prefer to start another PR for $ServiceToTask removal. The PR is to decouple the two parameter `$ServiceDirectory` and `$ServiceToTest`, which is convenient for the next step.

Reply:
Talked offline. As this is the only place to use $ServiceToTest, we'd like to remove in the same PR.
@heaths @JoshLove-msft 
We have an assumption that #3, #4 have no necessary to run against everything (*) in core pipeline. Do you have any concerns for the assumption?